### PR TITLE
Remove RBAC limitation from the token authentication

### PR DESCRIPTION
**Pull request by israel-hdez**
This is for plain Kubernetes environments, where the Projects API is not available. There was a RBAC limitation where Kiali was requiring read access to all Kiali accessible namespaces in order for users to be able to login.

This tries to circumvent the RBAC limiation by using the Kiali Service Account. Certainly, this is assuming that the Kiali ServiceAccount is present and has the `get namespaces` privilege. We are already assuming the presence of Kiali ServiceAccount with other sets of permissions for the following features:

* Kiali's cache
* Discovery of prometheus
* Discovery of Grafana
* Check what version of Istio's Telemetry is enabled

So, the assumption of the presence of the Kiali ServiceAccount and it's usage may not be too bad. Installation through Kiali Operator is currently creating the SA, at least with read-only privileges. Anyway, if the Kiali ServiceAccount is missing, or not properly mounted, or it doesn't have the assumed privileges, these changes should still fail gracefully.

Fixes kiali/kiali#2728

### CODE REVIEW

**Comment by israel-hdez** [comment]
BTW, I tested in OpenShift by hacking the hasProjects flag. I'm still pending to try on minikube.

File: business/namespaces.go [hunk 1]
```diff
@@ -98,18 +98,28 @@ func (in *NamespaceService) GetNamespaces() ([]models.Namespace, error) {
...
 					if errors.IsNotFound(err) {
+						// If a namespace is not found, then we skip it from the list of namespaces
 						log.Warningf("Kiali has an accessible namespace [%s] which doesn't exist", ans)
-					} else {
+					} else if !errors.IsForbidden(err) {
+						// Also, if namespace isn't readable, skip it.
+
+						// ...but if for some reason error is not "forbidden", abort and return the error.
 						return nil, err
 					}
```
[the remark this answers is not in the document]
Comment (israel-hdez) [reply business/namespaces.go]: It has the same result, but changing for code clarity.

File: business/namespaces.go [hunk 2]
```diff
@@ -209,3 +219,58 @@ func (in *NamespaceService) GetNamespace(namespace string) (*models.Namespace, e
 	}
 	return &result, nil
 }
+
+func (in *NamespaceService) getNamespacesUsingKialiSA(labelSelector string, forwardedError error) ([]core_v1.Namespace, error) {
+	// Check if we already are using the Kiali ServiceAccount token. If we do, no need to do further processing, since
```
[the remark this answers is not in the document]
Comment (israel-hdez) [reply business/namespaces.go]: Fixed.

**Comment by israel-hdez** [comment]
I just tested on minikube, and everything looks OK.

**Comment by pbajjuri20** [comment]
Thanks @israel-hdez,

Tested and Passed on Kiali UI 1.19.0 (dd31e321604369d816908006c386346f25996788).

**Comment by pbajjuri20** [comment]
Thanks @israel-hdez 

Tested and Passed! It looks good to me.

**Comment by israel-hdez** [comment]
Thanks @pbajjuri20. Sorry for all the headache.

**Pull request merged**